File: sdks/java/src/main/java/com/blok/blok/errors/ErrorCategory.java
package com.blok.blok.errors;

/**
 * Classifies the type of error that occurred during node execution.
 */
public enum ErrorCategory {

    /** Schema or input validation failure. */
    VALIDATION,

    /** Error during node execution. */
    EXECUTION,

    /** Misconfiguration of node or server. */
    CONFIGURATION,

    /** Network or connectivity issue. */
    NETWORK,

    /** Requested resource was not found. */
    NOT_FOUND
}

File: sdks/ruby/lib/blok/errors/error_category.rb
# frozen_string_literal: true

module Blok
  module Errors
    # ErrorCategory classifies the type of error that occurred during node execution.
    module ErrorCategory
      VALIDATION    = "VALIDATION"
      EXECUTION     = "EXECUTION"
      CONFIGURATION = "CONFIGURATION"
      NETWORK       = "NETWORK"
      NOT_FOUND     = "NOT_FOUND"

      ALL = [VALIDATION, EXECUTION, CONFIGURATION, NETWORK, NOT_FOUND].freeze
    end
  end
end
